Design/Instrument-2.0/ScanningInstruments_Design.md

```diff
@@ -0,0 +1,102 @@
+## Scanning Instrument Support for the ILL
+
+### Background
+
+#### Instrument 2.0
+
+The full discussion on Instrument 2.0 is beyond the scope of this document, more details are contained in this [requirements document](requirements-v2.md). A short summary on [scanning instrument requirements is given](requirements-v2.md#moving-instruments-requirements) within this document.
+
+For the purposes of the discussion here scanning instruments (or moving instruments) describe instruments that have detectors that can be in different positions during a single run. This forms a step scan. This is used to cover data collection where there are gaps between detectors. The aim of the work is to be able to deal with such detectors that move during a run, and be able to accumulate the data into a single workspace with the detectors at different positions throughout the run. This does not include two other use cases from the Instrument 2.0 requirements document; continuous scans and triple axis spectrometers.
+
+#### ILL Requirements
+
+The ILL have a requirement to implement support for moving instruments by the end of April 2017, in order to work with some of the diffractometers in use there. It is planned to implement the Instrument 2.0 support for moving instruments early in Mantid, to support the work at the ILL in a way that will not need to change with the full implementation of instrument 2.0. This document aims to describe the necessary steps to implement the Instrument 2.0 parts required.
+
+#### Overview of Proposed Implementation
+
+Two classes related to this have already been implemented in Mantid, [`SpectrumInfo`](https://github.com/mantidproject/mantid/pull/17394) and [`DetectorInfo`](https://github.com/mantidproject/mantid/pull/17875). These define the new way of working with commonly used information about a spectrum, namely isMasked, isMonitor, l2, twoTheta, signedTwoTheta, position, rotation and eFixed (coming soon). Eventually `DetectorInfo` will be responsible for the ownership of the parameters for the position and rotation, so for example moving a detector would have the new position stored in the `DetectorInfo` class, instead of the parameter map as is done currently. Values such as `l2` and `twoTheta` would use the new position when called and cache the return value, until another position change invalidates it.
+
+Support for the moving instruments will be implemented via the `DetectorInfo` class. The `SpectrumInfo` class needs no direct knowledge of the time indexing. The methods on the `DetectorInfo` class for accessing detector information will be accessed by giving a detector index and an optional time index. It will also hold information on the start and end times for each time index. Any further information, for example to normalise to monitor counts, can be held in a time series sample log.
```

Comment:
**Usage** of the `SpectrumInfo` class needs no direct knowledge of the time indexing.